gif/Form1.cs

```diff
@@ -179,5 +191,80 @@ private void lstInfectedFiles_DoubleClick(object sender, EventArgs e)
                 System.Diagnostics.Process.Start("explorer.exe", args);
             }
         }
+
+        private void bwrLoadAddonList_DoWork(object sender, System.ComponentModel.DoWorkEventArgs e)
+        {
+            updateAddonList();
+        }
+
+        private void btnUpdateIDs_Click(object sender, EventArgs e)
+        {
+            updateAddonList();
+        }
+
+        private void updateAddonList()
+        {
+            try
+            {
+                lblScannedFile.Invoke(new MethodInvoker(delegate 
+                {
+                    lblScannedFile.Text = "Attempting download of latest infected IDs";
+                }));
+
+                // Create a HttpResponse<essage whhich will get the raw .txt file contents
+                message = _client.GetAsync(_infectedAddonListURL).Result;
+
+                // Ensure that the request goes through with Status Code 200, else throw an exception
+                message.EnsureSuccessStatusCode();
+
+                // Get the retrieved contents and convert it into a string[] array through splitting the string
+                // The result is all the ID's separated by a new line
+                knownListOfInfectedAddonIDs = message.Content.ReadAsStringAsync().Result.Split('\n');
+
+                lblScannedFile.Invoke(new MethodInvoker(delegate
+                { 
+                    displaySuccess("Download successful!");
+                }));
+            }
+            catch (IOException ioEx)
+            {
+                lblScannedFile.Invoke(new MethodInvoker(delegate
+                {
+                    displayError($"There was an error trying to access/modify the file: {ioEx.Message}");
+                }));
+                knownListOfInfectedAddonIDs = backupListOfIDs;
+                return;
+            }
+            catch (UnauthorizedAccessException uaEx)
+            {
+                lblScannedFile.Invoke(new MethodInvoker(delegate
+                {
+                    displayError($"This program does not have access to the program's location: {uaEx.Message}");
+                }));
+                knownListOfInfectedAddonIDs = backupListOfIDs;
+                return;
+            }
+            catch (Exception genEx)
+            {
+                lblScannedFile.Invoke(new MethodInvoker(delegate
+                {
+                    displayError($"There was an error trying to use the program: {genEx.Message}");
```

Comment:
Shouldn't have the text displayed on the lblScannedFile since messages will overflow outside of the window. Instead use a MessageBox.Show or something equivalent.

Reply:
I'll go correct that, I was thinking of originally changing those message boxes I originally added